File: sdk-node/examples/contacts/create.ts
/**
 * Create a new contact. Fails with 409 if the email is already in use for
 * this host.
 *
 * Run:
 *   npx tsx examples/contacts/create.ts
 */

import { ThreeCommon, ThreeCommonConflictError } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

try {
  const created = await client.contacts.create({
    email: 'guest@example.com',
    firstName: 'Alex',
    lastName: 'Garcia',
  })
  console.log(`created ${created.id} <${created.email}>`)
} catch (err) {
  if (err instanceof ThreeCommonConflictError) {
    console.warn('contact with that email already exists for this host')
  } else {
    throw err
  }
}

File: sdk-python/examples/contacts/create.py
"""Run with: python examples/contacts/create.py

Fails with ``ConflictError`` if a contact with the same email already exists
for this host.
"""

from __future__ import annotations

from threecommon import ConflictError, ThreeCommon
from threecommon.contacts import CreateBody


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        try:
            created = client.contacts.create(
                CreateBody(
                    email="guest@example.com",
                    first_name="Alex",
                    last_name="Garcia",
                )
            )
            print(f"created {created.id} <{created.email}>")
        except ConflictError:
            print("contact with that email already exists for this host")


if __name__ == "__main__":
    main()
